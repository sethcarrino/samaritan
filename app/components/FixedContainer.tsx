import React, { ReactNode } from 'react';
import { View, StyleSheet, ViewStyle } from 'react-native';
import { DEVICE_WIDTH } from '../styles/globals';

interface Props {
  children: ReactNode;
}

const FixedContainer: React.FC<Props> = ({ children }) => (
  <View style={styles.fixedContainer}>{children}</View>
);

type FixedContainerStyleSheet = {
  fixedContainer: ViewStyle;
};

const styles = StyleSheet.create<FixedContainerStyleSheet>({
  fixedContainer: {
    position: 'absolute',
    bottom: 0,
    height: 120,
    width: DEVICE_WIDTH,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#fff',
    borderTopColor: 'lightgrey',
    borderTopWidth: 1
  }
});

export default FixedContainer;
