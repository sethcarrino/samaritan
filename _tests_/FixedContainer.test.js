import React from 'react';
import FixedContainer from '../app/components/FixedContainer';

import renderer from 'react-test-renderer';

describe('<FixedContainer/>', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<FixedContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
