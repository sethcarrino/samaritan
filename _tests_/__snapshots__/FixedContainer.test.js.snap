// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FixedContainer/> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderTopColor": "lightgrey",
      "borderTopWidth": 1,
      "bottom": 0,
      "height": 120,
      "justifyContent": "center",
      "position": "absolute",
      "width": 750,
    }
  }
/>
`;
